src/Main/Character/CharacterSelecter.js

```diff
@@ -0,0 +1,117 @@
+import React  from 'react';
+import { withRouter } from 'react-router-dom';
+import PropTypes from 'prop-types';
+import SelectSearch from 'react-select-search';
+import REALMS from 'common/REALMS';
+import makeUrl from './makeUrl';
+
+class CharacterSelecter extends React.PureComponent {
+
+  static propTypes = {
+    history: PropTypes.shape({
+      push: PropTypes.func.isRequired,
+    }),
+  };
+
+  constructor() {
+    super();
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.state = {
+      currentRegion: 'EU',
+      currentRealm: '',
+    };
+  }
+
+  componentDidMount() {
+    if (this.regionInput) {
+      this.regionInput.focus();
+    }
+  }
+
+  handleSubmit(e) {
+    e.preventDefault();
+
+    const region = this.regionInput.value;
+    const realm = this.state.currentRealm;
+    const char = this.charInput.value;
+
+    if (!region || !realm || !char) {
+      alert('Please select a region, realm and player.');
+      return;
+    }
+
+    //check if character has an key in localStorage, if so directly redirect to /character otherwise ask bnet-api
+    //checking here makes it more userfriendly and saves WCL-requests when char doesn't even exist for the bnet-api
+    const image = localStorage.getItem(`${region}/${realm}/${char}`);
+    if (image) {
+      this.props.history.push(makeUrl(region, realm, char));
+      return;
+    }
+
+    this.setState({ loading: true });
+    return fetch(`https://${region}.api.battle.net/wow/character/${encodeURIComponent(realm)}/${encodeURIComponent(char)}?locale=en_GB&apikey=n6q3eyvqh2v4gz8t893mjjgxsf9kjdgz`)
```

Comment:
Both are technically necessary. Someone could share the `wowanalyzer.com/character/REGION/REALM/CHARACTER/`-URL and don't have the image that gets parsed by `CharacterSelecter.js` and stored into the localStorage.
Fetching there in the first place is done because the WCL-API can take quite some time (up to 6-7sec based on amount of parses) to response and the request needs to be done anyways to get the character-render. This also reduces the amount of unnecessary WCL requests when we know that a character doesn't exist on WoW-API.